daffodil-core/src/main/scala/org/apache/daffodil/grammar/AlignedMixin.scala

```diff
@@ -202,9 +202,15 @@ trait AlignedMixin extends GrammarMixin { self: Term =>
       val parentAlignmentApprox =
```

Comment:
Yeah, I did not attempt to fix that issue. I don't think we've come across any formats that run into this issue, especially since most formats actually have really simple alignment--either all bits or all bytes. My guess is that's why it's marked as minor. 

My goal with with this PR was to just get the issues with alignment deadlocks resolved that were affecting real world schemas that Mike found. This alignment algorithm is complex enough that I'm sure there are some edge cases we just don't handle correctly, some we probably don't even know about yet.

Reply:
Not in this PR, but for users I'm quite sure we're still going to need the switches to "just shut the d*mn alignment stuff off" for them. 

It is beyond state-of-the-art in data format description to help people debug and give sensible easily followed diagnostics, in these sorts of circular or near-circular alignment relationships. "Just shut off alignment" is a perfectly acceptable response from users to our currently minimal attempts to provide diagnostics.

Reply:
Agreed, I've added [DAFODIL-2652](https://issues.apache.org/jira/browse/DAFFODIL-2652) to add the ability to disable alignment checking/parsers/unparsers.